scripts/cmake/common.cmake

```diff
@@ -74,45 +74,47 @@ if(CMAKE_CXX_COMPILER_ID MATCHES "GNU|Clang")
     add_compiler_flags(-Werror)
     add_compiler_flags(-fstrict-aliasing)
 
-    # The following options are not valid when clang-cl is used.
-    if(NOT MSVC)
-        add_compiler_flags(-pedantic)
-        add_compiler_flags(-pedantic-errors)
-        add_compiler_flags(-fvisibility=hidden)
-    endif()
+    # # The following options are not valid when clang-cl is used.
+    # if(NOT MSVC)
+    #     add_compiler_flags(-pedantic)
+    #     add_compiler_flags(-pedantic-errors)
+    #     add_compiler_flags(-fvisibility=hidden)
+    # endif()
```

Comment:
Sorry for the delay. Why is disabling these flags necessary?

Reply:
Some of the gcc flags are not known to clang-tidy and it complain it
The compile_comands.json generated from cake file is parced by clang-tidy.

My hope is -Wpedantic activate them all